meilisearch/src/lib.rs

```diff
@@ -289,7 +289,14 @@ fn import_dump(
         let mut index_reader = index_reader?;
         let metadata = index_reader.metadata();
         log::info!("Importing index `{}`.", metadata.uid);
-        let index = index_scheduler.create_raw_index(&metadata.uid)?;
+        let date;
+        if let Some(dump_reader_date) = dump_reader.date() {
+            date = Some((dump_reader_date, dump_reader_date));
+        } else {
+            date = Some((time::OffsetDateTime::now_utc(), time::OffsetDateTime::now_utc()));
+        }
+
+        let index = index_scheduler.create_raw_index(&metadata.uid, date)?;
```

Comment:
Once again, we're not interested in knowing the current time or at which date the dump was created but when the **index** was created.

You must use the date contained in the index, not anywhere else, and if you don't have any date, don't try to autogenerate one just send `None` to the `create_raw_index` function.

Reply:
where is the date field located in the index? There's tons of items that contain the word "index" and also plenty of search results for "date"

Reply:
In this part of the code, the only variable related to the index is the `index_reader`; if you check its implementation, you'll see it contains only three functions, one to get the documents, one to get the settings and one to get the metadata. And here is what the metadata looks like:
```rust
#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct IndexMetadata {
    pub uid: String,
    pub primary_key: Option<String>,
    #[serde(with = "time::serde::rfc3339")]
    pub created_at: OffsetDateTime,
    #[serde(with = "time::serde::rfc3339")]
    pub updated_at: OffsetDateTime,
}
```

Reply:
I'm trying to learn the process by which I could figure this out if I didn't have access to someone more experienced.

So [`index` variable is created as the output of `create_raw_index`](https://github.com/meilisearch/meilisearch/blob/99253094929745037ca63dcce73e3fafdf731e8f/meilisearch/src/lib.rs#L292). In the implementation, the `index_reader` seems to be [an element of iterator created from dump_reader.indexes()](https://github.com/meilisearch/meilisearch/blob/99253094929745037ca63dcce73e3fafdf731e8f/meilisearch/src/lib.rs#L288).

But the `indexes` method is part of `V6Reader`, which has plenty of functions and not just three. What should I do differently in my problem-solving process to arrive at the places you're referring to?